cli/src/potpie/server_manager.py

```
@@ -0,0 +1,391 @@
+
+            logging.error(
+                "PostgreSQL check failed: %s",
+                result.stderr.strip() or result.stdout.strip(),
+            )
+            raise PostgresError(
+                "PostgreSQL check failed: %s"
+                % (result.stderr.strip() or result.stdout.strip())
+            )
+        except subprocess.CalledProcessError as e:
+            logging.error(f"Failed to check PostgreSQL status: {e}")
+            raise PostgresError("Failed to check PostgreSQL status")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)